nearswap/src/storage_management.rs

```diff
@@ -0,0 +1,110 @@
+        if registration_only {
+            // Registration only setups the account but doesn't leave space for tokens.
+            if self.deposits.contains_key(&account_id) {
+                if amount > 0 {
+                    Promise::new(env::predecessor_account_id()).transfer(amount);
+                }
+            } else {
+                let refund = amount - min_balance;
+                if refund > 0 {
+                    Promise::new(env::predecessor_account_id()).transfer(refund);
+                }
+
+                let _acc_deposit = AccountDeposit {
+                    near: 0,
+                    storage_used: 0,
+                    tokens: HashMap::new()
+                };
+                self.deposits.insert(&account_id, &_acc_deposit);
+            }
+        } else {
+            self.deposit_near();
+        }
+        self.storage_balance_of(account_id.try_into().unwrap())
+            .unwrap()
+    }
+
+    #[allow(unused_variables)]
+    fn storage_withdraw(&mut self, amount: Option<U128>) -> StorageBalance {
+        assert_one_yocto();
+        let account_id = env::predecessor_account_id();
+        let account_deposit = self
+            .deposits
+            .get(&account_id)
+            .expect(ERR20_ACC_NOT_REGISTERED);
+        
+        // storagre available
+        let available = account_deposit.near - account_deposit.storage_usage();
+        let amount = if let Some(a) = amount { a.0 } else { available };
+        assert!(amount <= available, ERR14_NOT_ENOUGH_NEAR_DEPOSITED);
+        Promise::new(account_id.clone()).transfer(amount);
+        self.storage_balance_of(account_id.try_into().unwrap())
+            .unwrap()
+    }
+
+    #[allow(unused_variables)]
+    fn storage_unregister(&mut self, force: Option<bool>) -> bool {
+        assert_one_yocto();
+        let account_id = env::predecessor_account_id();
+        if let Some(account_deposit) = self.deposits.get(&account_id) {
+            assert!(
+                account_deposit.tokens.is_empty(),
+                "ERR_STORAGE_UNREGISTER_TOKENS_NOT_EMPTY"
+            );
+            self.deposits.remove(&account_id);
+            Promise::new(account_id.clone()).transfer(account_deposit.near);
+            true
+        } else {
+            false
+        }
+    }
+
+    fn storage_balance_bounds(&self) -> StorageBalanceBounds {
+        StorageBalanceBounds {
+            min: U128(MIN_ACCOUNT_DEPOSIT_LENGTH * (env::storage_byte_cost())),
+            max: None,
+        }
+    }
+
+    // check if a user is registered by calling
+    fn storage_balance_of(&self, account_id: ValidAccountId) -> Option<StorageBalance> {
+        let _acc_deposits = self
+            .deposits
+            .get(account_id.as_ref())
+            .unwrap_or(return None);
+        return Some(StorageBalance {
+            total: U128(_acc_deposits.near),
+            available: U128(_acc_deposits.near - _acc_deposits.storage_usage()),
+        })
+    }
```

Comment:
Please add unit tests for this function.

Reply:
ok